src/api/authentication.js

```diff
@@ -36,108 +34,20 @@ const auditingExemptPaths = [
-  const credentials = basicAuth(ctx)
-  if (credentials == null) {
-    // No basic auth details found
-    return null
-  }
-  const {name: email, pass: password} = credentials
-  const user = await UserModelAPI.findOne({
-    email: caseInsensitiveRegex(email)
-  })
-  if (user == null) {
-    // not authenticated - user not found
-    ctx.throw(
-      401,
-      `No user exists for ${email}, denying access to API, request originated from ${ctx.request.host}`,
-      {email}
-    )
-  }
-
-  const hash = crypto.createHash(user.passwordAlgorithm)
-  hash.update(user.passwordSalt)
-  hash.update(password)
-  if (user.passwordHash !== hash.digest('hex')) {
-    // not authenticated - password mismatch
-    ctx.throw(
-      401,
-      `Password did not match expected value, denying access to API, the request was made by ${email} from ${ctx.request.host}`,
-      {email}
-    )
-  }
-  return user
-}
-
-async function authenticateToken(ctx) {
-  const {header} = ctx.request
-  const email = header['auth-username']
-  const authTS = header['auth-ts']
-  const authSalt = header['auth-salt']
-  const authToken = header['auth-token']
-
-  // if any of the required headers aren't present
-  if (
-    isUndefOrEmpty(email) ||
-    isUndefOrEmpty(authTS) ||
-    isUndefOrEmpty(authSalt) ||
-    isUndefOrEmpty(authToken)
-  ) {
-    ctx.throw(
-      401,
-      `API request made by ${email} from ${ctx.request.host} is missing required API authentication headers, denying access`,
-      {email}
-    )
-  }
-
-  // check if request is recent
-  const requestDate = new Date(Date.parse(authTS))
-
-  const authWindowSeconds =
-    config.api.authWindowSeconds != null ? config.api.authWindowSeconds : 10
-  const to = new Date()
-  to.setSeconds(to.getSeconds() + authWindowSeconds)
-  const from = new Date()
-  from.setSeconds(from.getSeconds() - authWindowSeconds)
-
-  if (requestDate < from || requestDate > to) {
-    // request expired
-    ctx.throw(
-      401,
-      `API request made by ${email} from ${ctx.request.host} has expired, denying access`,
-      {email}
-    )
-  }
-
-  const user = await UserModelAPI.findOne({
-    email: caseInsensitiveRegex(email)
-  })
-  if (user == null) {
-    // not authenticated - user not found
-    ctx.throw(
-      401,
-      `No user exists for ${email}, denying access to API, request originated from ${ctx.request.host}`,
-      {email}
-    )
-  }
-
-  const hash = crypto.createHash('sha512')
-  hash.update(user.passwordHash)
-  hash.update(authSalt)
-  hash.update(authTS)
-
-  if (authToken !== hash.digest('hex')) {
-    // not authenticated - token mismatch
-    ctx.throw(
-      401,
-      `API token did not match expected value, denying access to API, the request was made by ${email} from ${ctx.request.host}`,
-      {email}
-    )
-  }
+async function authenticateBasic(ctx, next) {
+  // Basic auth using middleware
+  await passport.authenticate('basic', function (err, user) {
```

Comment:
The function here takes in a callback. It does return a promise we can `await`

Reply:
next is also not being used here

Reply:
well noted, will be updated